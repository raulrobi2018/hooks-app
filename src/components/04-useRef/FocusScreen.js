import React, {useRef} from "react";
import "./styles.css";

export const FocusScreen = () => {
    // El useRef sirve para hacer referencia a un elemento dado
    const inputRef = useRef();

    const handleClick = () => {
        // El atributo current es devuelto por el useRef,
        // en este caso va a ser el elemento input al cual le estoy haciendo referencia
        inputRef.current.focus();
    };

    return (
        <div>
            <h1>Focus Screen</h1>
            <hr />

            <input
                // Referencia al useRef
                ref={inputRef}
                type="text"
                placeholder="Name"
                className="form-control"
            ></input>

            <button
                className="btn btn-outline-primary mt-3"
                onClick={handleClick}
            >
                Focus
            </button>
        </div>
    );
};
