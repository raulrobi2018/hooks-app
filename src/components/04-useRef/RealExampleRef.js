import React, {useState} from "react";
import {MultipleCustomHooks} from "../03-examples/MultipleCustomHooks";
import "./styles.css";

export const RealExampleRef = () => {
    const [show, setShow] = useState(false);

    return (
        <>
            <h1>Real Example useRef</h1>
            <hr />

            {/* Si show est√° en true, muestra el componente, en caso contrario lo oculta */}
            {show && <MultipleCustomHooks />}

            <button
                className="btn btn-secondary mt-3"
                onClick={() => {
                    setShow(!show);
                }}
            >
                Show/Hide
            </button>
        </>
    );
};
