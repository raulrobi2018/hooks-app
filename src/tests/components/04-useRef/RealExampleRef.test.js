import React from "react";
import { RealExampleRef } from "../../../components/04-useRef/RealExampleRef";
import { shallow } from "enzyme";

//Para que me de la ayuda de funciones hago esta importaciÃ³n
import "@testing-library/jest-dom";

jest.mock("../../../hooks/useFetch");

describe("Testing RealExampleRef", () => {
  const wrapper = shallow(<RealExampleRef />);

  test("should display correctly", () => {
    expect(wrapper).toMatchSnapshot();
    //El componente no debe existir cuando arranca
    expect(wrapper.find("MultipleCustomHooks").exists()).toBe(false);
  });

  test("should display the MultipleCustomHook component", () => {
    wrapper.find("button").simulate("click");
    expect(wrapper.find("MultipleCustomHooks").exists()).toBe(true);
  });
});
