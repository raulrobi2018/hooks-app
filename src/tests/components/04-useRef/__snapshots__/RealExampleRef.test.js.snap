// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing RealExampleRef should display correctly 1`] = `
<Fragment>
  <h1>
    Real Example useRef
  </h1>
  <hr />
  <button
    className="btn btn-secondary mt-3"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</Fragment>
`;
